rtCommon/bidsArchive.py

```diff
@@ -654,3 +723,89 @@ def getIncremental(self, imageIndex: int = 0, **entities) \
+        """
+        Get a BIDS Run from the archive.
+
+        Args:
+            entities: Entities defining a run in the archive.
+
+        Returns:
+            A BidsRun containing all the BidsIncrementals in the specified run.
+
+        Raises:
+            NoMatchError: If the entities don't match any runs in the archive.
+            QueryError: If the entities match more than one run in the archive.
+
+        Examples:
+            >>> archive = BidsArchive('/tmp/dataset')
+            >>> run = archive.getBidsRun(subject='01', session='02',
+                                         task='test-task', run=1)
+            >>> print(run.numIncrementals())
+            53
+        """
+        images = self.getImages(**entities)
+        if len(images) == 0:
+            raise NoMatchError(f"Found no runs matching entities {entities}")
+        if len(images) > 1:
+            entities = [img.get_entities() for img in images]
+            raise QueryError("Provided entities were not unique to one run; "
+                             "try specifying more entities "
+                             f" (got runs with these entities: {entities}")
+        else:
+            bidsImage = images[0]
+            niftiImage = bidsImage.get_image()
+            niftiData = getNiftiData(niftiImage)
+            metadata = self.getSidecarMetadata(bidsImage)
+            metadata.pop('extension')  # only used in PyBids
+
+            run = BidsRun(**bidsImage.get_entities())
+            for imageIdx in range(niftiImage.header.get_data_shape()[3]):
+                newData = niftiData[..., imageIdx]
+                newImage = nib.Nifti1Image(newData,
+                                           niftiImage.affine,
+                                           niftiImage.header)
+
+                newIncremental = BidsIncremental(newImage, metadata)
+                run.appendIncremental(newIncremental, validateAppend=False)
+
+            return run
+
+    def appendBidsRun(self, run: BidsRun) -> None:
+        """
+        Append a BIDS Run to this archive.
+
+        Args:
+            run: Run to append to the archvie.
+
+        Examples:
+            >>> archive1 = BidsArchive('/tmp/dataset1')
+            >>> archive2 = BidsArchive('/tmp/dataset2')
+            >>> archive1.getRuns()
+            [1, 2]
+            >>> archive2.getRuns()
+            [1]
+            >>> run2 = archive1.getBidsRun(subject='01', task='test', run=2)
+            >>> archive2.appendBidsRun(run2)
+            >>> archive2.getRuns()
+            [1, 2]
+        """
+        numIncrementals = run.numIncrementals()
+        if numIncrementals == 0:
+            return
+
+        self.appendIncremental(run.asSingleIncremental(), useCache=False)
+
+    def flushCache(self) -> None:
+        """
+        Flush the cache by writing back to disk, if necessary, and updating the
+        BIDSLayout view of disk, if necessary.
+        """
```

Comment:
I feel like we need a __del__ function that would call flushCache() when the bidsArchive is deleted.